python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -1160,6 +1161,11 @@ def key(self) -> AssetCheckKey:
         return AssetCheckKey(asset_key=self.asset_key, name=self.name)
 
 
+# SYSTEM_METADATA_KEY_ASSET_EXECUTION_TYPE used to live on the metadata of external assets and
```

Comment:
If you are looking for places to break this up, separating the external data changes are a good place to look